nemo/utils/exp_manager.py

```diff
@@ -101,6 +101,48 @@ class ExpManagerConfig:
     checkpoint_callback_params: Optional[CallbackParams] = CallbackParams()
     # Additional exp_manager arguments
     files_to_copy: Optional[List[str]] = None
+    # logs timing of train/val/test steps
+    log_step_timing: Optional[bool] = True
+    # if True torch.cuda.synchronize() is called on start/stop
```

Comment:
This should possibly be a nested dict with following defaults (represented as a Data class, modifiable via Hydra) -

step_timing_kwargs:
  reduction: "mean" 
  sync_cuda: False

Then the instantiation of the timer insider expmanager can be TimerCallback(**cfg.step_timing_kwargs)

This further allows more aarguments + keep exp manager cleaner

Reply:
Great suggestion! Done.